WordPressKit/WordPressKit/ReaderTopicServiceRemote+Subscription.swift

```diff
@@ -73,25 +73,25 @@ extension ReaderTopicServiceRemote {
     ///   - frequency: The frequency value
     ///   - success: Success block
     ///   - failure: Failure block
-    @nonobjc public func updateFrequencyPostsEmail(with siteId: NSNumber, frequency: ReaderServiceDeliveryFrequency, _ success: @escaping () -> Void, _ failure: @escaping (NSError?) -> Void) {
+    @nonobjc public func updateFrequencyPostsEmail(with siteId: Int, frequency: ReaderServiceDeliveryFrequency, _ success: @escaping () -> Void, _ failure: @escaping (ReaderTopicServiceError?) -> Void) {
         let parameters = [Delivery.frequency: NSString(string: frequency.rawValue)]
         POST(with: .postsEmail(siteId: siteId, action: .update), parameters: parameters, success: success, failure: failure)
     }
     
     
     // MARK: Private methods
     
-    private func POST(with request: ReaderTopicServiceSubscriptionsRequest, parameters: [String: AnyObject]? = nil, success: @escaping () -> Void, failure: @escaping (NSError?) -> Void) {
+    private func POST(with request: ReaderTopicServiceSubscriptionsRequest, parameters: [String: AnyObject]? = nil, success: @escaping () -> Void, failure: @escaping (ReaderTopicServiceError?) -> Void) {
         let urlRequest = path(forEndpoint: request.path, withVersion: request.apiVersion)
         
         DDLogInfo("URL: \(urlRequest)")
         
         wordPressComRestApi.POST(urlRequest, parameters: parameters, success: { (_, response) in
             DDLogInfo("Success \(response?.url?.absoluteString ?? "unknown response url")")
             success()
-        }) { (error, _) in
+        }) { (error, response) in
             DDLogError("Error: \(error.localizedDescription)")
-            failure(error)
+            failure(ReaderTopicServiceError.remoteResponse(message: error.localizedDescription, url: response?.url?.absoluteString ?? "unknown response url"))
```

Comment:
For easier readability could `response?.url?.absoluteString ?? "unknown response url"` be performed first and the result passed as an argument? 
Also, we might consider shortening the message to "unknown url" but either way we need to localize the string (NSLocalizedString).